infra/cifuzz/run_fuzzers.py

```diff
@@ -0,0 +1,156 @@
+
+  Args:
+    fuzz_seconds: The total time allotted for fuzzing.
+    workspace: The location in a shared volume to store a git repo and build
+      artifacts.
+    project_name: The name of the relevant OSS-Fuzz project.
+    sanitizer: The sanitizer the fuzzers should be run with.
+
+  Returns:
+    (True if run was successful, True if bug was found).
+  """
+  # Validate inputs.
+  if not os.path.exists(workspace):
+    logging.error('Invalid workspace: %s.', workspace)
+    return False, False
+
+  logging.info('Using %s sanitizer.', sanitizer)
+
+  out_dir = os.path.join(workspace, 'out')
+  artifacts_dir = os.path.join(out_dir, 'artifacts')
+  os.makedirs(artifacts_dir, exist_ok=True)
+  if not fuzz_seconds or fuzz_seconds < 1:
+    logging.error('Fuzz_seconds argument must be greater than 1, but was: %s.',
+                  fuzz_seconds)
+    return False, False
+
+  # Get fuzzer information.
+  fuzzer_paths = utils.get_fuzz_targets(out_dir)
+  if not fuzzer_paths:
+    logging.error('No fuzzers were found in out directory: %s.', out_dir)
+    return False, False
+
+  # Run fuzzers for allotted time.
+  total_num_fuzzers = len(fuzzer_paths)
+  fuzzers_left_to_run = total_num_fuzzers
+  min_seconds_per_fuzzer = fuzz_seconds // total_num_fuzzers
+  for fuzzer_path in fuzzer_paths:
+    run_seconds = max(fuzz_seconds // fuzzers_left_to_run,
+                      min_seconds_per_fuzzer)
+
+    target = fuzz_target.FuzzTarget(fuzzer_path,
+                                    run_seconds,
+                                    out_dir,
+                                    project_name,
+                                    sanitizer=sanitizer)
+    start_time = time.time()
+    testcase, stacktrace = target.fuzz()
+    fuzz_seconds -= (time.time() - start_time)
+    if not testcase or not stacktrace:
+      logging.info('Fuzzer %s, finished running.', target.target_name)
+    else:
+      utils.binary_print(b'Fuzzer %s, detected error:\n%s' %
+                         (target.target_name.encode(), stacktrace))
+      shutil.move(testcase, os.path.join(artifacts_dir, 'test_case'))
+      parse_fuzzer_output(stacktrace, artifacts_dir)
+      return True, True
+    fuzzers_left_to_run -= 1
+
+  return True, False
+
+
+def parse_fuzzer_output(fuzzer_output, out_dir):
+  """Parses the fuzzer output from a fuzz target binary.
+
+  Args:
+    fuzzer_output: A fuzz target binary output string to be parsed.
+    out_dir: The location to store the parsed output files.
+  """
+  # Get index of key file points.
+  for marker in STACKTRACE_TOOL_MARKERS:
+    marker_index = fuzzer_output.find(marker)
+    if marker_index:
+      begin_summary = marker_index
+      break
+
+  end_summary = -1
+  for marker in STACKTRACE_END_MARKERS:
+    marker_index = fuzzer_output.find(marker)
+    if marker_index:
+      end_summary = marker_index + len(marker)
+      break
+
+  if begin_summary is None or end_summary is None:
```

Comment:
nit: instead of begin_summary and end_summary, might be better to say begin_stack and end_stack ?